report_calculator.py

```diff
@@ -0,0 +1,53 @@
+import statistics
+
+
+def get_month_attr_values(month_data, attr):
+    """Returns readings of a given attribute on given month 
+    """
+
+    month_attr_values = [(day_data[attr],
+                          day_data[list(day_data.keys())[0]])
+                         for day_data in month_data
+                         if day_data[attr] is not None]
+    return month_attr_values
+
+
+def get_attr_values(weather_dataset, attr, year, month=-1):
+    """Returns readings of a given attribute on given year/month 
+    """
+
+    attr_values = []
+    year_data = weather_dataset[year]
+    if month == -1:
+        for month in year_data:
+            attr_values.extend(get_month_attr_values(year_data[month], attr))
+    else:
+        attr_values = get_month_attr_values(year_data[month], attr)
+
+    return attr_values
+
+
+def get_value(weather_dataset, attr, value_type, year, month=-1):
+    """calculates the provided value on given year and month
+    """
+
+    attr_values = get_attr_values(weather_dataset, attr, year, month)
+    if value_type == "min":
+        value_index = min(range(len(attr_values)), key=attr_values.__getitem__)
+    elif value_type == "max":
+        value_index = max(range(len(attr_values)), key=attr_values.__getitem__)
```

Comment:
@ahmadx16 Still duplicating two line :( 
```suggestion
    limit_value_function = min if value_type == "min" else max
    value_index = limit_value_function(range(len(attr_values)), key=attr_values.__getitem__)```